lib/upgrade/etcd/conversionv1v3/node.go

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conversionv1v3
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+
+	apiv1 "github.com/projectcalico/libcalico-go/lib/apis/v1"
+	"github.com/projectcalico/libcalico-go/lib/apis/v1/unversioned"
+	apiv3 "github.com/projectcalico/libcalico-go/lib/apis/v3"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	cnet "github.com/projectcalico/libcalico-go/lib/net"
+)
+
+type Node struct{}
+
+// convertKVPairToAPI converts a KVPair containing a backend Node and NodeKey
+// to an API Node structure.
+// This is part of the conversionHelper interface.
+func (_ Node) BackendV1ToAPIV3(d *model.KVPair) (Resource, error) {
```

Comment:
What's the plan for populating `Node.Spec.NodeBGPSpec.IPv4IPIPTunnelAddr`?

@tmjd @robbrockbank?

Reply:
That'll be retro-fitted by the upgrade script, but for simple YAML conversion it won't be possible to include that field.

Reply:
I added a comment saying that it needs to be populated external to the function.